WordPress/WordPressTest/Services/PostAutoUploadInteractorTests.swift

```diff
@@ -31,10 +31,12 @@ class PostCoordinatorUploadActionUseCaseTests: XCTestCase {
             createPost(.publish, confirmedAutoUpload: true): .upload,
             // Published local drafts with no confirmation will be remote auto-saved
             createPost(.publish): .autoSave,
-            // Draft and published posts with remote are currently unsupported. This will be
-            // fixed soon.
-            createPost(.draft, hasRemote: true): .nothing,
-            createPost(.publish, hasRemote: true): .nothing,
+            // Posts with remote that do not have confirmation will be remote auto-saved
```

Comment:
I completely understand the desire of adding more test cases with less-effort but in this specific case, I had to understand the *logic* of this test in order to understand it.

In general, I'm in favor of "DAMP (Descriptive And Meaningful Phrases) not DRY (Don't repeat yourself)" in tests. Usually, I thought of the tests as documentation, and I'd like the reader to understand what the test is doing as quickly as possible, without thinking too much — and if there's duplicated code in the process I really don't mind.

Take the first case as an example, it will be like this without the parametrization:

```
    func testReturnsUploadAsActionForDraftedPost() {
        let post = createPost(.draft)

        let action = interactor.autoUploadAction(for: post)

        expect(action).to(equal(.upload))
    }
```

What do you think? I found it much more expressive.

We would end up with a lot of test functions? Probably. But I tend to find them more meaningful than the comments.

Anyway, this is just an addition to an existed test. So, I don't think we need to change it right now. We can save for a discussion in the near future. :)

Reply:
I think we've discussed this in the previous PR and I understand your argument. 

> I had to understand the logic of this test in order to understand it

It could be that I'm used to parameterized tests that I don't notice this. I'm curious to know how you _read_ this code and made it difficult to understand. Was it the use of dictionary? Was it the variable name (`postsAndExpectedActions`) that was not clear? Which logic did you find was difficult to understand? Is there something that we can do to make it easier for you? 

What did you think of the `Expectation` struct? I concocted that with you in mind. I knew you were gonna comment on this part. 🙂 

https://github.com/wordpress-mobile/WordPress-iOS/blob/2c6380a68b583dafe3eba8bb7cccaa69790bbf5b/WordPress/WordPressTest/ViewRelated/Post/Utils/PostNoticeViewModelTests.swift#L28-L34

> We would end up with a lot of test functions? Probably. But I tend to find them more meaningful than the comments.

That's true. If I was going to write this that way, I might still end up making a code generator for it. 😉 Or what do you think about grouping them by statuses? One method for `.draft` cases, and another for `.publish`, and so on.

Reply:
Discussed tests with Leandro earlier today. Marking this as resolved.